image/descriptor.go

```diff
@@ -50,17 +51,26 @@ func (d *descriptor) hash() string {
 
 func listReferences(w walker) (map[string]*descriptor, error) {
 	refs := make(map[string]*descriptor)
+	var index v1.ImageIndex
 
 	if err := w.walk(func(path string, info os.FileInfo, r io.Reader) error {
-		if info.IsDir() || !strings.HasPrefix(path, "refs") {
+		if info.IsDir() || filepath.Clean(path) != "index.json" {
 			return nil
 		}
 
-		var d descriptor
-		if err := json.NewDecoder(r).Decode(&d); err != nil {
+		if err := json.NewDecoder(r).Decode(&index); err != nil {
 			return err
 		}
-		refs[info.Name()] = &d
+
+		var d descriptor
+		for i := 0; i < len(index.Manifests); i++ {
+			if index.Manifests[i].Descriptor.Annotations["org.opencontainers.ref.name"] != "" {
+				d.MediaType = index.Manifests[i].Descriptor.MediaType
+				d.Digest = string(index.Manifests[i].Descriptor.Digest)
+				d.Size = index.Manifests[i].Descriptor.Size
+				refs[index.Manifests[i].Descriptor.Annotations["org.opencontainers.ref.name"]] = &d
```

Comment:
I try to use `type descriptor v1.Descriptor` instead of descriptor, but the following error occurs:

> /tmp/go-build341304265/github.com/opencontainers/image-tools/image/_test/_obj_test/descriptor.go:47:cannot use d.Digest (type digest.Digest) as type string in argument to strings.SplitN
/tmp/go-build341304265/github.com/opencontainers/image-tools/image/_test/_obj_test/descriptor.go:81: cannot use &index.Manifests[i].Descriptor (type *v1.Descriptor) as type *descriptor in assignment

What I am using is what I can think of.As for the kind of ideas you say I want to do, but can not solve the problem after the amendment.